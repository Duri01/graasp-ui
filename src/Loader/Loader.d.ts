import React from 'react'
import { LoadingType } from 'react-loading'

interface LoaderProps {
  type: LoadingType
  color: string
  className?: string
}

declare const Loader: React.FC<LoaderProps>
