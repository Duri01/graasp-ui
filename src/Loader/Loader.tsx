import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';

export interface LoaderProps {
  color?: string;
  className?: string;
}

export const Loader: React.FC<LoaderProps> = ({ className }) => (
  <div className={className}>
    <CircularProgress color='primary' />
  </div>
);

export default Loader;
