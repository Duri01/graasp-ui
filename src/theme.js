export const PRIMARY_COLOR = '#5050d2'
