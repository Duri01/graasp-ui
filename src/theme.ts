export const PRIMARY_COLOR = '#5050d2';
export const SECONDARY_COLOR = '#FFFFFF';
