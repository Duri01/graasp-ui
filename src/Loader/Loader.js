import React from 'react'
import ReactLoading from 'react-loading'
import { PRIMARY_COLOR } from '../theme'

const Loader = ({ type = 'bubbles', color = PRIMARY_COLOR, className }) => (
  <div className={className}>
    <ReactLoading type={type} color={color} />
  </div>
)

export default Loader
