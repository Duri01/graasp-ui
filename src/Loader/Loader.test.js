import React from 'react';
import ReactLoading from 'react-loading';
import { shallow } from 'enzyme';
import { Loader } from './Loader';
import { PRIMARY_COLOR } from '../theme';

const createLoaderProps = ({ type, color, className } = {}) => {
  return {
    className,
    type,
    color,
  };
};

describe('Loader', () => {
  let wrapper;

  describe('default values', () => {
    beforeAll(() => {
      const props = createLoaderProps();
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<Loader {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <ReactLoading /> component with default values', () => {
      const reactLoading = wrapper.find(ReactLoading);
      expect(reactLoading).toHaveLength(1);
      expect(reactLoading.prop('type')).toEqual('bubbles');
      expect(reactLoading.prop('color')).toEqual(PRIMARY_COLOR);
    });
  });

  describe('prop values', () => {
    beforeAll(() => {
      const props = createLoaderProps({
        type: 'sometype',
        color: 'red',
        className: 'someclass',
      });
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<Loader {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <ReactLoading /> component with prop values', () => {
      const reactLoading = wrapper.find(ReactLoading);
      expect(reactLoading).toHaveLength(1);
      expect(reactLoading.prop('type')).toEqual('sometype');
      expect(reactLoading.prop('color')).toEqual('red');
      const divWrapper = wrapper.find('div.someclass');
      expect(divWrapper).toHaveLength(1);
    });
  });
});
