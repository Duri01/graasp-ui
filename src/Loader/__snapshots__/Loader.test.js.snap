// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader default values renders correctly 1`] = `
<div>
  <Loading
    color="#5050d2"
    delay={0}
    height={64}
    type="bubbles"
    width={64}
  />
</div>
`;

exports[`Loader prop values renders correctly 1`] = `
<div
  className="someclass"
>
  <Loading
    color="red"
    delay={0}
    height={64}
    type="sometype"
    width={64}
  />
</div>
`;
